components/console-backend-service/internal/domain/k8s/namespace_service.go

```diff
@@ -71,6 +76,33 @@ func (svc *namespaceService) Create(name string, labels gqlschema.Labels) (*v1.N
 	return svc.client.Namespaces().Create(&namespace)
 }
 
+func (svc *namespaceService) Update(name string, labels gqlschema.Labels) (*v1.Namespace, error) {
+	var updated *v1.Namespace
+	err := retry.RetryOnConflict(retry.DefaultBackoff, func() error {
+		namespace, err := svc.Find(name)
+
+		if err != nil {
+			return errors.Wrapf(err, "while getting %s [%s]", pretty.Namespace, name)
+		}
+		if namespace == nil {
+			return apiErrors.NewNotFound(schema.GroupResource{
+				Group:    "",
+				Resource: "namespaces",
+			}, name)
+		}
+		namespace.ObjectMeta.Labels = labels
+
+		updated, err = svc.client.Namespaces().Update(namespace)
+
+		return errors.Wrapf(err, "while updating %s [%s]", pretty.Namespace, name)
+	})
+
+	if err == nil {
```

Comment:
Just to keep go convention I would switch the `if` branches so it would be `err != nil` here.